app/Http/Controllers/LockerClaimController.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Carbon\Carbon;
+use App\Models\Client;
+use App\Models\Locker;
+use App\Models\LockerClaim;
+use Illuminate\Support\Str;
+use Illuminate\Http\Request;
+use App\Http\Resources\LockerClaimResource;
+use App\Exceptions\NotYetImplementedException;
+use App\Http\Requests\LockerClaimStoreRequest;
+use App\Http\Requests\LockerClaimUpdateRequest;
+
+class LockerClaimController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @param  string  $lockerGuid
+     * @return \Illuminate\Http\Response
+     */
+    public function index(string $lockerGuid)
+    {
+        $locker = Locker::where('guid', $lockerGuid)->firstOrFail();
+        $lockerClaims = $locker->claims;
+
+        return LockerClaimResource::collection($lockerClaims);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  string  $lockerGuid
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(string $lockerGuid, LockerClaimStoreRequest $request)
+    {
+        $locker = Locker::where('guid', $lockerGuid)->firstOrFail();
+
+        $email = $request->get('email');
+        $client = Client::where('email', $email)->firstOrCreate([
+            'email' => $email,
+        ]);
+
+        $startMoment = $request->get('start_at', Carbon::now());
+        $endMoment = $request->get('end_at', Carbon::now()->addDays(7));
+
+        $startMomentParsed = Carbon::parse($startMoment);
+        $endMomentParsed = Carbon::parse($endMoment);
+
+        if (!$locker->isCurrentlyAvailable()) {
+            return response()->json([
+                'message' =>
+                    'The locker is not available, because it is already ' .
+                    'claimed somewhere between ' . $startMomentParsed .
+                    ' and ' . $endMomentParsed . '.',
+            ], 400);
+        }
+
+        $lockerClaim = $client->lockerClaims()->create([
```

Comment:
Should there be a reference in here to the Client (for email usage later)?

Reply:
Because it is created through the `$client` model,  it adds it automatically.